# test

In [5]:
def max_test(imgfile,value=0):
    import SimpleITK as sitk
    eq = sitk.ReadImage(imgfile)
    eq_array = sitk.GetArrayFromImage(eq)
    import numpy as np
    if value==0:
        eq_max = eq_array.max()
    else:
        eq_max = value
    print(eq_max, np.argwhere(eq_array==eq_max))
max_test('./test/112200_046080_048450.tif_eq80.tif_i_1_mask.tif',144)
max_test('./test/112200_046080_048450.tif_eq80.tif',144)
max_test('./test/112200_046080_048450.tif')

144 [[  4 212 179]]
144 [[  4 212 179]
 [135 104 232]
 [137 101 210]
 [160 102 188]
 [160 115 185]
 [160 116 186]
 [160 123 189]
 [161 112 183]
 [161 125 181]
 [162 103 185]
 [162 118 189]
 [162 119 189]
 [162 123 188]
 [162 124 187]
 [162 128 180]
 [162 128 181]
 [163 108 185]
 [163 109 186]
 [163 111 180]
 [163 121 190]
 [163 122 188]
 [163 123 187]
 [163 123 189]
 [163 128 180]
 [163 129 179]
 [163 130 180]
 [164 104 180]
 [164 104 183]
 [164 108 188]
 [164 110 178]
 [164 117 189]
 [164 119 188]
 [164 127 177]
 [165 104 181]
 [165 105 179]
 [165 109 187]
 [165 113 189]
 [165 121 187]
 [165 128 176]
 [166 105 177]
 [166 106 185]
 [167  58 222]
 [174  42 174]
 [178  71 201]
 [179 126 156]
 [181 170 156]
 [182 170 155]
 [184 127 150]
 [188 175 145]
 [194  69 157]
 [194 185 143]
 [195  69 157]
 [195 169 177]
 [196  69 157]
 [196  74 157]
 [196 125 140]
 [197  75 156]
 [198  70 158]
 [202 134  78]
 [205 226  87]
 [212 141 108]
 [221  54 161]
 [226 136  89]
 [230 132  89]]
1328 [[161 110 

# debug on gs1891-size128

In [6]:
import os,glob
image_file_dir = '/PBshare/SEU-ALLEN/Users/XiaoqinGu/datatest'
image_file_list = glob.glob(image_file_dir+'/*/*/*.tif')
for image_file in image_file_list:
    msd_str = f'/media/lyx/OldMan/MSD-C/msd -i {image_file}'
    print(msd_str)
    #os.system(msd_str)
    break

/media/lyx/OldMan/MSD-C/msd -i /PBshare/SEU-ALLEN/Users/XiaoqinGu/datatest/18470/2/new.tif


input imgfile is /PBshare/SEU-ALLEN/Users/XiaoqinGu/datatest/18470/2/new.tif
raw tif channel/frame/height/width/datatype/stripsize: -116981744/128/128/128/21920/8192
*** Error in `/media/lyx/OldMan/MSD-C/msd': free(): invalid next size (normal): 0x0000000001efd9e0 ***
======= Backtrace: =========


input imgfile is test/112200_046080_048450.tif
raw tif channel/frame/height/width/datatype/stripsize: 16/255/255/256/6649/130560
raw tif max voxel value: 1328 and location: 161 110 185
raw tif min voxel value: 16 and location: 31 169 224


# test on gs1891-raw256